app/src/main/java/com/cornellappdev/android/eateryblue/ui/components/login/LoginPage.kt

```diff
@@ -15,181 +19,150 @@ import androidx.compose.material.Button
-                focusRequester = passwordFocus,
-                isPassword = true,
-                onSubmit = {
-                    if (password.isNotEmpty()) {
-                        focusManager.clearFocus()
-                    }
-                }
-            )
-
-            Button(
-                enabled = clickable,
-                shape = RoundedCornerShape(24.dp),
-                modifier = Modifier
-                    .fillMaxWidth()
-                    .padding(top = 66.dp)
-                    .height(56.dp),
-                onClick = {
+        CustomTextField(
+            modifier = Modifier.then(shimmerModifier),
+            value = loginState.password,
+            onValueChange = {
+                loginViewModel.onPasswordTyped(it)
+            },
+            enabled = !loginState.loading,
+            placeholder = "Type your password...",
+            backgroundColor = GrayZero,
+            focusRequester = passwordFocus,
+            isPassword = true,
+            onSubmit = {
+                if (loginState.password.isNotEmpty()) {
                     focusManager.clearFocus()
-                    attemptSignIn = true
-                },
-                colors = ButtonDefaults.buttonColors(
-                    backgroundColor = if (clickable) EateryBlue else GrayZero
-                ),
-                elevation = ButtonDefaults.elevation(defaultElevation = 0.dp)
-            ) {
-                Text(
-                    text = if (attemptSignIn) "Logging in..." else "Log in",
-                    color = if (clickable) Color.White else GrayThree,
-                    style = EateryBlueTypography.h5
-                )
+                }
             }
+        )
+        Button(
+            enabled = clickable,
+            shape = RoundedCornerShape(24.dp),
+            modifier = Modifier
+                .fillMaxWidth()
+                .then(shimmerModifier)
+                .padding(top = 66.dp)
+                .height(56.dp),
+            onClick = {
+                focusManager.clearFocus()
+                loginViewModel.onLoginPressed()
+            },
+            colors = ButtonDefaults.buttonColors(
+                backgroundColor = if (clickable) EateryBlue else GrayZero
+            ),
+            elevation = ButtonDefaults.elevation(defaultElevation = 0.dp)
+        ) {
+            Text(
+                text = if (loginState.loading) "Logging in..." else "Log in",
+                color = if (clickable) Color.White else GrayThree,
+                style = EateryBlueTypography.h5
+            )
         }
-
-        if (attemptSignIn) {
+        if (loginState.loading) {
             // LoginWebView should take up no space whatsoever. User should not be able to see it.
             Box(
                 modifier = Modifier
                     .size(0.dp)
-                    .zIndex(-2f)
+                    .zIndex(-1f)
             ) {
```

Comment:
Since you made the `LoginWebView` invisible (which is a lot better than this lol) just delete this so people dont think this works / is necessary lol